src/components/UI/Overlay/Overlay.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { useState } from "react";
+import css from "./Overlay.module.scss";
+import { SvgIcon, SvgIconX, SvgIconSearch } from "../../UI/SvgIcon";
+import { CsvUploadButton } from "../../AddressBook/CsvUploadButton";
+import { AddressBook } from "../../../common/hooks/useAddressBook";
+import { makeEtherscanAddressURL } from "../../../utils";
+import { isEmpty, pickBy } from "lodash";
+
+interface OverlayProps {
+  id?: string;
+  isOverlayVisible: boolean;
```

Comment:
Usually visibility is controlled by the parent component, when it's not visible it wont even render. Something like:

```
const OverlayParent = () => {
...
  return(
    <Something /> 
    {showOverlay && <Overlay />}
    <SomethingElse /> 
  )
}
```

Is there a reason you chose to let the child decide to show or now?

Reply:
the above method would be a show/hide treatment with no transitions.

i've implemented CSSTransition from react instead, to better manage transitions when components mount and unmount